numpy/core/src/multiarray/ctors.c

```diff
@@ -2498,6 +2498,11 @@ PyArray_FromInterface(PyObject *origin)
     if (ret == NULL) {
         goto fail;
     }
+    /*
+     * Ref to dtype was stolen by PyArray_NewFromDescrAndBase
+     * Prevent DECREFing dtype in fail codepath by setting to NULL
+     */
+    dtype = NULL;
```

Comment:
Thanks, this should go before the error check, since in that case the goto fail also happens.

Reply:
fixed

Reply:
But does `PyArray_NewFromDescrAndBase` steal a reference to the dtype when it fails allocating the array? That sounds unlikely to me.

Reply:
In any of the paths where `PyArray_NewFromDescrAndBase` fails (returns `NULL`) it `DECREF`s the dtype.